src/Utils.php

```diff
@@ -23,6 +23,8 @@
  * @author Odín del Río <odin.drp@gmail.com>
  *
  * @internal
+ *
+ * @deprecated This is a God Class antipattern. Don't extend it. It is fine to use logic that is already here [so we don't trigger deprecation warnings], but over time logic should move to dedicated, single-responsibility classes.
```

Comment:
```suggestion
 * @deprecated This is a God Class anti-pattern. Don't expand it. It is fine to use logic that is already here (that's why we don't trigger deprecation warnings), but over time logic should be moved to dedicated, single-responsibility classes.
```

I've changed `extend` to `expand` because it could be perceived as something about inheritance.